kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/Tls.java

```diff
@@ -29,4 +54,18 @@ public static String getStoreTypeOrPlatformDefault(String storeType) {
     public boolean definesKey() {
         return key != null;
     }
+
+    @Override
+    public TlsClientAuth clientAuth() {
+        // We can't just default this in the canonical constructor as it then becomes a nonsensical property of Upstream TLS config
+        return clientAuth == null ? TlsClientAuth.NONE : clientAuth;
```

Comment:
Tom and I were chatting about defaults for a bit as well. 

The current round trip test is enforcing a stronger guarantee than we need. I was attempting to do the minimum in this pr. rather than bolt on a refactoring there.

Initially I defaulted null to none in the constructor but that broke configuration tests as the none was serialised back out which means it appears in the upstream tls nodes which is meaningless at best.